source/packages/libraries/core/version/scripts/compile.ts

```diff
@@ -0,0 +1,48 @@
+import fs from 'fs';
+import readline from 'readline';
+import { Project } from 'ts-morph';
+
+async function extractVersion(): Promise<string> {
+    const versionPoliciesReadStream = fs.createReadStream(
+        '../../../../common/config/rush/version-policies.json'
+    );
+    const rl = readline.createInterface({
+        input: versionPoliciesReadStream,
+        crlfDelay: Infinity,
+    });
+
+    let previousLine: string = '';
+    let versionLine: string | undefined = undefined;
+    for await (const line of rl) {
+        if (previousLine.includes('"policyName": "external",')) {
+            versionLine = line;
+            break;
+        }
+
+        previousLine = line;
+    }
+
+    // https://semver.org/#is-there-a-suggested-regular-expression-regex-to-check-a-semver-string
+    const versionRegex =
+        /(0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*)(?:-((?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+([0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?/;
+    const version = versionLine!.match(versionRegex);
+
+    return version![0];
+}
+
+async function compile() {
+    const emittingProject = new Project({
+        tsConfigFilePath: 'tsconfig.json',
+    });
+    const version = await extractVersion();
+    console.log(`BENJI: in compile: version: ${version}`);
```

Comment:
Remove log (or make it clearer and use the logger tool and debug level).

Reply:
Good catch!